07-routing-gossip.md

```diff
@@ -379,6 +379,20 @@ The node creating `channel_update` SHOULD accept HTLCs which pay an
 older fee for some time after sending `channel_update` to allow for
 propagation delay.
 
+## Pruning the Network View
+
+Nodes SHOULD monitor the funding transactions in the blockchain to identify channels that are being closed.
+If the funding output of a channel is being spent, then the channel is to be considered closed and SHOULD be removed from the local network view.
+
+Nodes MAY prune nodes added through `node_announcement` messages from their local view if the announced node no longer has any open channels associated.
+This is a direct result from the dependency of a `node_announcement` being preceded by a `channel_announcement`,
+
+Several scenarios may result in channels becoming unusable and the endpoints unable to send updates for these channels.
+This happens for example in case that both endpoints lose access to their private keys, and cannot sign a `channel_update` or close the channel on-chain.
+These channels are unlikely to be part of a computed route since they would be partitioned off from the rest of the network, however they would remain in the local network view, and be forwarded to other nodes forever.
```

Comment:
For a channel to become garbage we both endpoints have to die, and lose all their information (settlement tx and/or private key), otherwise they could close normally. For that channel to become reachable there must be a path leading to that channel, and someone along that channel is the first person that doesn't suffer that loss, so hopefully they'd close the channel that goes into the cluster of failed channels, and partition them off the network. It might be a bit wishful thinking that this'd happen right away but hopefully we could isolate this dysfunctional cluster just by this.